rust/lance-index/src/vector/ivf/builder.rs

```diff
@@ -40,8 +41,17 @@ pub struct IvfBuildParams {
 
     pub sample_rate: usize,
 
+    /// Precomputed partitions file (row_id -> partition_id)
+    /// mutually exclusive with `precomputed_shuffle_buffers`
     pub precomputed_partitons_file: Option<String>,
 
+    /// Precomputed shuffle buffers (row_id -> partition_id, pq_code)
+    /// mutually exclusive with `precomputed_partitons_file`
+    /// requires `centroids` to be set
+    ///
+    /// The input is expected to be (/dir/to/buffers, [buffer1.lance, buffer2.lance, ...])
```

Comment:
As a user I have no idea what the format of these files should be or what this feature is expected to do.  This can be an internal feature for now but we probably want to document this better at some point (probably need a whole doc on using the GPU)

Reply:
we don't expect most user to know these apis. This is very much "advanced mode"